src/main/scala/is/hail/annotations/aggregators/TransformedRegionValueAggregator.scala

```diff
@@ -0,0 +1,74 @@
+package is.hail.annotations.aggregators
+
+import is.hail.asm4s._
+import is.hail.expr._
+import is.hail.expr.ir._
+import is.hail.annotations._
+
+object TransformedRegionValueAggregator {
+  /**
+    * {@code t} is the type of value to which
+    * TransformedRegionValueAggregator.seqOp will be applied
+    *
+    * The argument to {@code makeTransform} is an IR that evalutes to the
+    * aggregable carrier struct. The output is an IR that evaluates to the
+    * desired input to {@code next.seqOp}.
+    *
+    **/
+  def apply(t: Type,
+    makeTransform: IR => IR,
+    next: RegionValueAggregator): TransformedRegionValueAggregator = {
+
+    val transform = makeTransform(In(0, t))
+    val out = makeCarrier(transform)
+    val fb = FunctionBuilder.functionBuilder[MemoryBuffer, Long, Boolean, Long]
+    Compile(out, fb)
+    new TransformedRegionValueAggregator(fb.result(), transform.typ, next)
+  }
+
+  private def carryingField = "x"
```

Comment:
This also makes me quite nervous.  Evaluating scalar aggregations should not create any intermediate garbage in the region.

Reply:
I agree, a key design goal is the avoidance of _real_ per-entry region garbage.

Yes, this creates per-entry region garbage, however, our previously discussed implementation strategy reclaims the region suffix used for per-entry operations.

Returning a containing struct is, AFAIK, currently the only way to return a possibly missing value from a compiled IR.

Reply:
> Yes, this creates per-entry region garbage, however, our previously discussed implementation strategy reclaims the region suffix used for per-entry operations.

I'm willing to _allow_ it based on the previous discussion, but it certainly shouldn't be necessary in the common, scalar case.

I still don't know how TransformedRVA is intended to be used, but perhaps my complaint is about the interface between aggregator caller and callee.  I'm hoping for an architecture where nobody has to return anything: the caller does the preprocessing (as we discussed above for filter, map and flatMap), and then calls the aggregator.  I don't see where anyone needs to return anything.

Reply:
Yes that is the architecture we should strive for. Right now I have no way to express, in the IR, a) an argument that is a Java object (e.g. an aggregator) and b) invoke a function on such an object. So, for this PR, I run the IR to produce a value which I, in turn, pass to the region value aggregator.

Probably the most straightforward way to get this working is to provide a different hook into the `Compiler` that returns the triplet of `Code` (the precompute, the value, and the missingness bit) instead of emitting it, and then I can write the Code corresponding to invoking the TransformedRegionValueAggregator. I'll add that to this PR.

Reply:
Ok this is done